js/packages/openinference-instrumentation-openai/src/instrumentation.ts

```diff
@@ -334,17 +337,64 @@ function getLLMInputMessagesAttributes(
   body: ChatCompletionCreateParamsBase,
 ): Attributes {
   return body.messages.reduce((acc, message, index) => {
+    const messageAttributes = getChatCompletionInputMessageAttributes(message);
     const index_prefix = `${SemanticConventions.LLM_INPUT_MESSAGES}.${index}`;
-    acc[`${index_prefix}.${SemanticConventions.MESSAGE_CONTENT}`] = String(
-      message.content,
-    );
-    acc[`${index_prefix}.${SemanticConventions.MESSAGE_ROLE}`] = String(
-      message.role,
-    );
+    // Flatten the attributes on the index prefix
+    for (const [key, value] of Object.entries(messageAttributes)) {
+      acc[`${index_prefix}.${key}`] = value;
+    }
     return acc;
   }, {} as Attributes);
 }
 
+function getChatCompletionInputMessageAttributes(
+  message: ChatCompletionMessageParam,
+): Attributes {
+  const role = message.role;
+  const attributes: Attributes = {
+    [SemanticConventions.MESSAGE_ROLE]: role,
+  };
+  switch (role) {
+    case "user":
+      // Add the content only if it is a string
+      if (typeof message.content === "string")
+        attributes[SemanticConventions.MESSAGE_CONTENT] = message.content;
+      break;
+    case "assistant":
+      if (message.tool_calls) {
+        message.tool_calls.forEach((toolCall, index) => {
+          const toolCallIndexPrefix = `${SemanticConventions.MESSAGE_TOOL_CALLS}.${index}.`;
+          attributes[
+            toolCallIndexPrefix + SemanticConventions.TOOL_CALL_FUNCTION_NAME
+          ] = toolCall.function.name;
+          attributes[
+            toolCallIndexPrefix +
+              SemanticConventions.TOOL_CALL_FUNCTION_ARGUMENTS_JSON
+          ] = JSON.stringify(toolCall.function.arguments);
```

Comment:
Maybe should add an `if` statement just for safety, because the API does not guarantee that `function` will the only tool forever.
```suggestion
          if (toolCall.function) {
            attributes[
              toolCallIndexPrefix + SemanticConventions.TOOL_CALL_FUNCTION_NAME
            ] = toolCall.function.name;
            attributes[
              toolCallIndexPrefix +
                SemanticConventions.TOOL_CALL_FUNCTION_ARGUMENTS_JSON
            ] = toolCall.function.arguments;
          }
```

Reply:
yeah good call - got a false sense of confidence with the types